migrate.sh

```diff
@@ -42,18 +42,48 @@ echo "Source Directory: $source_dir"
 echo "Destination Directory: $destination_dir"
 echo "" # Blank line to separate from any failure output
 
-# Convert source and destination directories to absolute paths
-source_dir=$(readlink -f "$source_dir")
-destination_dir=$(readlink -f "$destination_dir")
+is_absolute_path() {
+    case "$1" in
+        /*) return 0 ;;  # Absolute path
+        ~/*) return 0 ;; # Absolute path with home directory
+        *) return 1 ;;   # Relative path
+    esac
+}
+
+# Convert source and destination directory to absolute path if relative
+if ! is_absolute_path "$source_dir"; then
+    source_dir=$(readlink -f "$source_dir") || { echo "Error: Failed to resolve source directory path, directory may not exist"; exit 1; }
+fi
```

Comment:
I don't see the need for the `is_absolute_path` check here, just the following seems sufficient, similarly for the destination dir. 
```suggestion
source_dir=$(readlink -f "$source_dir") || { echo "Error: Failed to resolve source directory path, directory may not exist"; exit 1; }
```